gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -318,10 +318,30 @@ local function GetSceneDataFromPlayer(ply)
 	return data
 end
 
+local function CloneDamageInfo(dmginfo)
+	return {
+		ammoType = dmginfo:GetAmmoType(),
+		attacker = dmginfo:GetAttacker(),
+		baseDamage = dmginfo:GetBaseDamage(),
+		damage = dmginfo:GetDamage(),
+		damageBonus = dmginfo:GetDamageBonus(),
+		damageCustom = dmginfo:GetDamageCustom(),
+		damageForce = dmginfo:GetDamageForce(),
+		damagePosition = dmginfo:GetDamagePosition(),
+		damageType = dmginfo:GetDamageType(),
+		inflictor = dmginfo:GetInflictor(),
+		maxDamage = dmginfo:GetMaxDamage(),
+		reportedPosition = dmginfo:GetReportedPosition(),
+		isBulletDamage = dmginfo:IsBulletDamage(),
+		isExplosionDamage = dmginfo:IsExplosionDamage(),
+		isFallDamage = dmginfo:IsFallDamage(),
+	}
+end
+
```

Comment:
Is it possible to keep the dmgInfo function structure, so we dont have to revert other code than this cloning?
Like
isBulletDamage = dmgInfo:IsBulletDamage()
IsBulletDamage = function (dmgInfo) return dmgInfo.isBulletDamage end

Reply:
i thought about it, but it's a lot of extra work -- IsBulletDamage is the same as `dmginfo:GetDamageType() == DMG_BULLET`, same with the rest of the `Is` functions

Reply:
Yeah I get that, but what I mean is, store our own table of dmginfo with private variables and then keep the functions referencing those cloned states.

Reply:
if you do not implement @ZenBre4ker s suggestion, you shuld probably rename the gmginfo table to something like `damageInfoData` or so because otherwise people might think this is a real damageinfo object. Which it isn't

Reply:
this makes sense, most of the existing consumers of the .dmginfo are probably still going to break with the properties being different so no sense in keeping it as dmginfo

Reply:
@ZenBre4ker what's your current stance on this? can we merge it?